app/controllers/catalog_controller.rb

```diff
@@ -1257,7 +1257,7 @@ def default_entry_point(prov_type, service_type)
   def get_form_vars
     copy_params_if_present(@edit[:new], params, %i[st_prov_type name description provision_cost catalog_id dialog_id generic_subtype long_description zone_id price retire_fqname reconfigure_fqname fqname])
 
-    @edit[:new][:display] = params[:display] == "1"
+    @edit[:new][:display] = params[:display] == "1" if params[:display] # Need to check also presence of params[:display] to prevent changing @edit[:new][:display]
```

Comment:
```suggestion
    @edit[:new][:display] = params.try(:[], :display) == "1" 
```

Reply:
The suggestion does not work as expected. `@edit[:new][:display]` should't be changed, if `params[:display]` is not set.

Reply:
The assignment itself needs to be under condition (as @hstastna has it) not just the right side.